cpp/src/arrow/array/builder_primitive.cc

```diff
@@ -105,12 +105,12 @@ Status BooleanBuilder::AppendValues(const uint8_t* values, int64_t length,
   return Status::OK();
 }
 
-Status BooleanBuilder::AppendValues(const std::vector<uint8_t>& values,
+Status BooleanBuilder::AppendValues(const util::span<uint8_t> values,
```

Comment:
So, `const std::vector<T>&` should be replaced with `util::span<const T>`, not `const util::span<T>` (because a `span`'s constness doesn't propagate to the elements it points to, unlike `vector`).